simplifier/operations/bitwise.py

```diff
@@ -175,6 +175,39 @@ def negate(self, level: Optional[int] = None, predecessors: Optional[List[Operat
         self.world.replace(self, new_op)
         return new_op
 
+    def get_equivalent_condition_of_negated_type(self, operation: Union[BitwiseAnd, BitwiseOr]) -> Union[BitwiseAnd, BitwiseOr]:
+        """
+        Create the new condition that is equivalent to self and replace self by it.
+
+        - if self is Or-operation and operation is a conjunction
+            self = (a1 & a2 & ... & ak) | l1 | l2 | ... | lh, where operation = (a1 & a2 & ... & ak),
+            remaining_conditions = [l1, l2, ..., lh] and each li is a condition
+            -> new_condition = (a1 | l1 | ... |lh) & (a2 | l1 | ... |lh) & ... & (ak | l1 | ... |lh)
+        -if self is And-operation and operation is a disjunction
+            self = (a1 | a2 | ... | ak) & l1 & l2 & ... & lh, where disjunction = (a1 | a2 | ... | ak),
+            remaining_conditions = [l1, l2, ..., lh] and each li is a condition
+            -> new_condition = (a1 & l1 & ... & lh) | (a2 & l1 & ... & lh) | ... | (ak & l1 & ... & lh)
+        """
+        conditions = []
+        for literal in operation.operands:
+            new_operands = [literal, *(cond.copy_tree() for cond in self.operands if hash(cond) != hash(operation))]
+            conditions.append(self.__class__(self.world).replace_operands(new_operands))
+        new_condition = self._negated_class(self.world).replace_operands(conditions)
+        for cond in new_condition.operands:
+            cond.simplify(keep_form=True)
+        self.world.replace(self, new_condition)
+        return new_condition
+
+    def get_junction(self):
+        """If the formula has a literal that is a junction of negated type, then return it, otherwise return None."""
+        for child in self.operands:
+            if isinstance(child, BitwiseNegate):
+                if resolved_neg := child.dissolve_negation():
+                    child = resolved_neg
+            if isinstance(child, self._negated_class):
+                return child
+        return None
```

Comment:
I think this function would be somewhat more generic as an iterator. Can there ever be only one operand of type `negated_class` or do we only care for the first one we find?

Reply:
We only want one, at least for the current use case. 
Given a formula such as `A | B | C | D` where `A, B, C, D` are also formulas and assume we want to bring it to CNF form. We search for one operand that is a conjunction, say B = b1 & b2 & b3. Now, we transform it into `(b1 | A | C | d) & (b2 | A | C | D) & (b3 | A | C | D)`. Now, we call to cnf to each operand. 

To summarize, currently, we really only want one, if one exists. However, there can be more than one but we only use the first we find. So an iterator could make it more complicated for the current use-case.